getoptpp.hpp

```diff
@@ -45,14 +49,15 @@ void Options::add(std::string const & name, int has_arg, int val, Handle handle)
 }
 
 void Options::add(std::string const & name, int has_arg, int* flag, int val, Handle handle) {
-	// create a new option from the args
-	options.emplace_back();
-	auto& option = options.back();
-	option.name = name.c_str();
-	option.has_arg = has_arg;
-	option.flag = flag;
-	option.val = val;
-	
+	char *name_copy = (char *)malloc(sizeof(char) * name.size());
```

Comment:
Why do you use malloc instead std::string? Btw imho, free is missing here, too.

My suggestion (not tested):
```c++
options.push_back({name.c_str(), has_arg, flag, val});
```
I'm not sure whether emplace_back would work - since the original `option` is a plain c struct (if I remember correctly). But I guess it wouldn't due to the lack of a suitable (or any^^) struct ctor.